Examples/Algorithms/TruthTracking/ActsExamples/TruthTracking/ParticleSmearing.hpp

```diff
@@ -27,28 +28,48 @@ namespace ActsExamples {
 class ParticleSmearing final : public BareAlgorithm {
  public:
   struct Config {
+    Config(const Options::Variables& vars) {
+      using namespace Acts::UnitConstants;
+      using Options::Reals;
+
+      auto sigmaD0Opts = vars["smear-sigma-D0"].template as<Reals<3>>();
+      auto sigmaZ0Opts = vars["smear-sigma-Z0"].template as<Reals<3>>();
+      auto sigmaMomOpts = vars["smear-sigma-momentum"].template as<Reals<3>>();
+
+      sigmaD0 = sigmaD0Opts[0] * Acts::UnitConstants::um;
+      sigmaD0PtA = sigmaD0Opts[1] * Acts::UnitConstants::um;
+      sigmaD0PtB = sigmaD0Opts[2] / Acts::UnitConstants::GeV;
+      sigmaZ0 = sigmaZ0Opts[0] * Acts::UnitConstants::um;
+      sigmaZ0PtA = sigmaZ0Opts[1] * Acts::UnitConstants::um;
+      sigmaZ0PtB = sigmaZ0Opts[2] / Acts::UnitConstants::GeV;
+      sigmaT0 = vars["smear-sigma-T0"].as<double>() * Acts::UnitConstants::ns;
+      sigmaPhi = sigmaMomOpts[0] * Acts::UnitConstants::degree;
+      sigmaTheta = sigmaMomOpts[1] * Acts::UnitConstants::degree;
+      sigmaPRel = sigmaMomOpts[2];
+    }
+
     /// Input truth particles collection.
     std::string inputParticles;
     /// Output smeared tracks parameters collection.
     std::string outputTrackParameters;
     /// Constant term of the d0 resolution.
-    double sigmaD0 = 30 * Acts::UnitConstants::um;
+    double sigmaD0;
     /// Pt-dependent d0 resolution of the form sigma_d0 = A*exp(-1.*abs(B)*pt).
-    double sigmaD0PtA = 0 * Acts::UnitConstants::um;
-    double sigmaD0PtB = 1 / Acts::UnitConstants::GeV;
+    double sigmaD0PtA;
+    double sigmaD0PtB;
     /// Constant term of the z0 resolution.
-    double sigmaZ0 = 30 * Acts::UnitConstants::um;
+    double sigmaZ0;
     /// Pt-dependent z0 resolution of the form sigma_z0 = A*exp(-1.*abs(B)*pt).
-    double sigmaZ0PtA = 0 * Acts::UnitConstants::um;
-    double sigmaZ0PtB = 1 / Acts::UnitConstants::GeV;
+    double sigmaZ0PtA;
+    double sigmaZ0PtB;
     /// Time resolution.
-    double sigmaT0 = 5 * Acts::UnitConstants::ns;
+    double sigmaT0;
     /// Phi angular resolution.
-    double sigmaPhi = 1 * Acts::UnitConstants::degree;
+    double sigmaPhi;
     /// Theta angular resolution.
-    double sigmaTheta = 1 * Acts::UnitConstants::degree;
+    double sigmaTheta;
     /// Relative momentum resolution.
-    double sigmaPRel = 0.001;
+    double sigmaPRel;
```

Comment:
Can we keep the source of truth for  the defaults in the config here?

Reply:
Ah should we? I mean default constructor doesn't exist anymore and it takes the default value from ParticleSmearingOptions.cpp

Reply:
Right but eventually we want to remove the boost program options. This has been a pain in the past because it was spread everywhere. I spent a non-trivial amount of time centralizing the defaults to these config structs as much as possible.

The python bindings and ideally the program options should take the defaults from this config struct, rather than the other way round.

Reply:
Okay I restored them but changed the default value to same with what was in `ReconstructionBase.cpp`